pkg/controller/metadata/metadata_controller.go

```diff
@@ -347,6 +349,44 @@ func (r *ReconcileMetadata) generateMetadataSecretData(instance *verifyv1beta1.M
 	return data, nil
 }
 
+// This function determins if we should regenerate the metadata or not.
+func ShouldRegenerate(secretsObj *corev1.Secret, hashOfSecrets string) bool {
```

Comment:
`hashOfSecrets` might be better named something like `hashOfRequestSpec`. Or something